File: src/main/java/org/mitallast/queue/node/Node.java
package org.mitallast.queue.node;

import com.google.inject.Injector;
import com.typesafe.config.Config;
import org.mitallast.queue.common.component.LifecycleComponent;

public interface Node extends LifecycleComponent {

    Config config();

    Injector injector();
}

File: src/main/java/org/mitallast/queue/node/Node.kt
package org.mitallast.queue.node

import com.google.inject.Injector
import com.typesafe.config.Config
import org.mitallast.queue.common.component.LifecycleComponent

interface Node : LifecycleComponent {

    fun config(): Config

    fun injector(): Injector
}
